cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tether/PeerInfo.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tether;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Peer information that's persisted in the tether store.
+ */
+public class PeerInfo {
+  private final String name;
+  private final String endpoint;
+  private final TetherStatus tetherStatus;
+  private final PeerMetadata metadata;
+  private final long lastConnectionTime;
+
+  public PeerInfo(String name, @Nullable String endpoint, TetherStatus tetherStatus, PeerMetadata metadata) {
```

Comment:
When will endpoint be null?

Reply:
Endpoint will be null on the server side